src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private User validUser;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        validUser = new User();
+        validUser.setLogin("bodje343r@gmail.com");
+        validUser.setPassword("121212");
+        validUser.setAge(27);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registeredValidUser_Ok() {
+        assertDoesNotThrow(() -> registrationService.register(validUser));
+    }
+
+    @Test
+    void registeredNullPassword_NotOk() {
+        validUser.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(validUser));
+    }
+
+    @Test
+    void registeredNullLoginUser_NotOk() {
+        validUser.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(validUser));
+    }
+
+    @Test
+    void registeredNullAge_NotOk() {
+        validUser.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(validUser));
+    }
+
+    @Test
+    void registeredShortLogin_NotOk() {
+        validUser.setLogin("Tom");
```

Comment:
try to test all edge cases

Reply:
same for password and age (do not create new tests, just add couple assertions to each test)

Reply:
One test still had to be added